EIPS/eip-5753.md

```diff
@@ -0,0 +1,230 @@
+        require(unlockers[id] == address(0), "ALREADY_LOCKED"); 
+        unlockers[id] = unlocker;
+        _approve(unlocker, id);
+    }
+
+    /**
+     * @dev Public function to unlock the token. Only the unlocker (stated at the time of locking) can unlock
+     */
+    function unlock(uint256 id) public virtual {
+        require(msg.sender == unlockers[id], "NOT_UNLOCKER");
+        unlockers[id] = address(0);
+    }
+
+    /**
+     * @dev Returns the unlocker for the tokenId
+     *      address(0) means token is not locked
+     *      reverts if token does not exist
+     */
+    function getLocked(uint256 tokenId) public virtual view returns (address) {
+        require(_exists(tokenId), "Lockable: locking query for nonexistent token");
+        return unlockers[tokenId];
+    }
+
+    /**
+     * @dev Locks the token
+     */
+    function _lock(address unlocker, uint256 id) internal virtual {
+        unlockers[id] = unlocker;
+    }
+
+    /**
+     * @dev Unlocks the token
+     */
+    function _unlock(uint256 id) internal virtual {
+        unlockers[id] = address(0);
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                              OVERRIDES
+    //////////////////////////////////////////////////////////////*/
+
+    function approve(address to, uint256 tokenId) public virtual override {
+        require (getLocked(tokenId) == address(0), "Can not approve locked token");
+        super.approve(to, tokenId);
+    }
+
+    function _beforeTokenTransfer(
+        address from,
+        address to,
+        uint256 tokenId
+    ) internal virtual override {
+        // if it is a Transfer or Burn
+        if (from != address(0)) { 
+            // token should not be locked or msg.sender should be unlocker to do that
+            require(getLocked(tokenId) == address(0) || msg.sender == getLocked(tokenId), "LOCKED");
+        }
+    }
+
+    function _afterTokenTransfer(
+        address from,
+        address to,
+        uint256 tokenId
+    ) internal virtual override {
+        // if it is a Transfer or Burn, we always deal with one token, that is startTokenId
+        if (from != address(0)) { 
+            // clear locks
+            delete unlockers[tokenId];
+        }
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                              ERC165 LOGIC
+    //////////////////////////////////////////////////////////////*/
+
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        virtual
+        override
+        returns (bool)
+    {
+        return
+            interfaceId == type(IERC721Lockable).interfaceId ||
+            super.supportsInterface(interfaceId);
+    }
+
+}
+```
+
+## Security Considerations
+The callers of `lock` function should always consider if there's an unlocking function in the contract, that is stated as an unlocker, unless they want to lock the NFT from transfers forever.
```

Comment:
This should be rephrased. Also, what if there's a bug that results in the NFT not being able to be unlocked?

Finally, I want to see considerations about MEV, and considerations about reentrancy by malicious EIP-5753 contracts.